llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -2264,6 +2263,14 @@ OpenMPIRBuilder::InsertPointOrErrorTy OpenMPIRBuilder::createSections(
     AfterIP = {FiniBB, FiniBB->begin()};
   }
 
+  // Now we can fix the dummy branch to point to the right place
+  if (!CancellationBranches.empty()) {
```

Comment:
Condition is unnecessary: If `CancellationBranches` is empty, the loop will not execute anyway